libraries/AP_NavEKF3/AP_NavEKF3_core.h

```diff
@@ -949,6 +982,7 @@ class NavEKF3_core : public NavEKF_core_common
     uint32_t airborneDetectTime_ms; // last time flight movement was detected
     Vector6 innovVelPos;            // innovation output for a group of measurements
     Vector6 varInnovVelPos;         // innovation variance output for a group of measurements
+    Vector6 velPosObs;              // observations for combined velocity and positon group of measurements (3x1 m , 3x1 m/s)
```

Comment:
@tridge says we should declare this differently so we can test we are not running past the end of the array.

Reply:
I thought that was what that type supported if MATH_CHECK_INDEXES ?

Reply:
Yes, I think you're right that it's using the correct type, thanks.

Reply:
yes, Vector6 is good. I was referring to places in the EKF where we do things like this:
  float H_DECL[24] = {};
that bypasses the index checking